lib/Remotery.c

```diff
@@ -5769,6 +5772,9 @@ static rmtError SampleThreadsLoop(rmtThread* rmt_thread)
                 Sample* sample = stalling_sample_tree.root->first_child;
                 assert(sample != NULL);
                 QueueSampleTree(thread_profilers->mqToRmtThread, sample, stalling_sample_tree.allocator, thread_profiler->threadName, thread_profiler, RMT_TRUE);
+
+                assert(stalling_sample_tree.allocator != NULL);
+                ObjectAllocator_Free(stalling_sample_tree.allocator, stalling_sample_tree.root);
```

Comment:
Can I verify that your understanding of this change is that:

1. `stalling_sample_tree.root->first_child` has been sent to the main Remotery thread via `QueueSampleTree`.
2. This will get released later when the Remotery thread has finished with it.
3. This leaves `stalling_sample_tree.root` allocated here and nobody frees it, leading to memory leak.

Your change here adds that required free call.

Reply:
This is correct, the `root` sample is leaking here.

Reply:
Great, could you leave a comment to that effect? The Sample Tree API needs cleaning up and this will help a later refactor.

Reply:
Sure. Left the comment in there.